dhall/src/Dhall/Normalize.hs

```diff
@@ -891,7 +905,7 @@ isNormalized e0 = loop (Syntax.denote e0)
       ToMap x t -> case x of
           RecordLit _ -> False
           _ -> loop x && all loop t
-      Field r k -> case r of
+      Field r (FieldSelection s0 k s1) -> Maybe.isNothing s0 && Maybe.isNothing s1 && case r of
```

Comment:
I think it would be slightly more readable to handle the `Maybe s` fields with two pattern matches, where the first is
```suggestion
      Field r (FieldSelection Nothing k Nothing) -> case r of
```